axelor-base/src/main/java/com/axelor/apps/base/service/administration/SequenceService.java

```diff
@@ -121,11 +121,17 @@ public static boolean isMonthValid(Sequence sequence) {
         && (seq.contains(PATTERN_YEAR) || seq.contains(PATTERN_FULL_YEAR));
   }
 
-  public static boolean isSequenceLengthValid(Sequence sequence) {
-    String seqPrefixe = StringUtils.defaultString(sequence.getPrefixe(), "").replace("%", "");
-    String seqSuffixe = StringUtils.defaultString(sequence.getSuffixe(), "").replace("%", "");
+  public void isSequenceLengthValid(Sequence sequence) throws AxelorException {
+    Company company = sequence.getCompany();
+    SequenceVersion sequenceVersion = getVersion(sequence, appBaseService.getTodayDate(company));
+    String nextSeq =
+        computeNextSeq(sequenceVersion, sequence, appBaseService.getTodayDate(company));
 
-    return (seqPrefixe.length() + seqSuffixe.length() + sequence.getPadding()) <= 14;
+    if (nextSeq.length() > 14) {
```

Comment:
Please add a new static variable SEQ_MAX_LENGTH instead of using directly the hard coded value